### PR TITLE
feature: search back-end

### LINKED ISSUE #18: Global search (with quick links)
Reference: https://ethz.ch/en.html

### PR DESCRIPTION
**Pull request by legitk8**
Partially fixes #18

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -32,3 +32,6 @@ out
 
 # Prisma
 prisma/generated/client/
+
+# typesense server data
+typesense-data/
```
Comment (maan2003) [inline .gitignore:37]: use your local git config `~/.config/git/ignore` globally or `.git/info/exclude` in repo

**Review by GetPsyched — CHANGES_REQUESTED** [review]
Rebase on master

File: .gitignore [hunk 2]
```diff
@@ -38,3 +38,16 @@ yarn-error.log*
 
 # typescript
 *.tsbuildinfo
+# Next.js build output
```
Comment (GetPsyched) [inline .gitignore:41]: Why is this in the diff?

File: search/search.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import Typesense, { Client } from 'typesense'
+import { SearchParams, SearchResponse } from 'typesense/lib/Typesense/Documents'
+import { CollectionMap, collectionMap, collections } from './collections'
+const query = 'arya'
```
Comment (GetPsyched) [inline search/search.ts:4]: Separate imports and declarations

**Review by GetPsyched — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 2 above)
Comment (GetPsyched) [reply .gitignore:41]: @legitk8 not resolved^

File: search/search.ts (hunk 3 above)
Comment (GetPsyched) [reply search/search.ts:4]: same here

**Review by GetPsyched — CHANGES_REQUESTED** [review]

File: server/typesense/schema.ts [hunk 4]
```diff
@@ -3,12 +3,12 @@ import { type CollectionCreateSchema } from 'typesense/lib/Typesense/Collections
...
   fields: [
-    { name: 'image', type: 'string', index: false },
+    { name: 'logo', type: 'string', index: false },
     { name: 'name', type: 'string' },
     { name: 'tagline', type: 'string' },
-    { name: 'admins', type: 'object[]', index: false, optional: true },
+    // { name: 'admins', type: 'object[]', index: false, optional: true },
   ],
-  default_sorting_field: 'name',
+  // default_sorting_field: 'name',
```
Comment (GetPsyched) [inline server/typesense/schema.ts:11]: Why is this commented out?

File: server/typesense/schema.ts [hunk 5]
```diff
@@ -20,21 +20,21 @@ const courseSchema: CollectionCreateSchema = {
...
-  default_sorting_field: 'code',
+  // default_sorting_field: 'code',
 };
 
 const facultyAndStaffSchema: CollectionCreateSchema = {
   name: 'faculty_and_staff',
   fields: [
     { name: 'employee_id', type: 'string', index: false },
-    { name: 'image', type: 'string', index: false },
+    { name: 'image', type: 'image', index: false },
```
Comment (GetPsyched) [inline server/typesense/schema.ts:30]: um no? it will be a URL which is a string.

File: server/typesense/init.ts [hunk 6]
```diff
@@ -1,6 +1,13 @@
 import { client, schema } from '.';
 
-schema.forEach(async (schema) => {
-  const generatedSchema = await client.collections().create(schema);
-  console.debug(generatedSchema);
-});
+// schema.forEach(async (schema) => {
+//   const generatedSchema = await client.collections().create(schema);
+//   console.debug(generatedSchema);
+// });
```
Comment (GetPsyched) [inline server/typesense/init.ts:6]: Why leave this in?

File: app/api/typesense-init/route.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import { selectAll } from "~/server/typesense/db_data";
+import { createCollectionsFromSchema } from "~/server/typesense/init";
+import { addAllData } from "~/server/typesense/data_init";
+import { resolve } from "path";
```
Comment (GetPsyched) [inline app/api/typesense-init/route.ts:4]: Unused import

**Review by GetPsyched — CHANGES_REQUESTED** [review]

**Comment by GetPsyched** [comment]
Done in #172

**Pull request closed without merging**